src/main/java/banking/dao/AccountDAO.java

```diff
@@ -0,0 +1,133 @@
+    }
+
+    public void insertCard(Card card) {
+        try (Connection connection = dbManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(INSERT_INTO_ACCOUNT)) {
+            preparedStatement.setString(1, card.getNumber());
+            preparedStatement.setString(2, card.getPIN());
+            preparedStatement.executeUpdate();
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    public boolean containCard(String cardNumber) {
+        try (Connection connection = dbManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SELECT_COUNT_NUMBER)) {
+            preparedStatement.setString(1, cardNumber);
+            ResultSet res = preparedStatement.executeQuery();
+            if (res.next()) {
+                return res.getBoolean(1);
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return false;
+    }
+
+    public Account login(String cardNumber, String PIN) {
+        try (Connection connection = dbManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SELECT_ACCOUNT_WHERE_NUMBER_AND_PIN)) {
+            preparedStatement.setString(1, cardNumber);
+            preparedStatement.setString(2, PIN);
+            ResultSet res = preparedStatement.executeQuery();
+            if (res.next()) {
+                return new Account(
+                        res.getInt("id"),
+                        new Card(res.getString("number"), res.getString("pin")),
+                        res.getInt("balance")
+                );
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return null;
+    }
+
+    public int getBalance(int id) {
+        int balance = -1;
+        try (Connection connection = dbManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_ACCOUNT_WHERE_ID)) {
+            preparedStatement.setInt(1, id);
+            ResultSet res = preparedStatement.executeQuery();
+            if (res.next()) {
+                balance = res.getInt("balance");
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return balance;
+    }
+
+    public String getNumber(int id) {
+        String number = "";
+        try (Connection connection = dbManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_ACCOUNT_WHERE_ID)) {
+            preparedStatement.setInt(1, id);
+            ResultSet res = preparedStatement.executeQuery();
+            if (res.next()) {
+                number = res.getString("number");
+            }
+        } catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        return number;
+    }
+
+    public void addBalance(int id, int amount) {
```

Comment:
The `addBalance()` and `updateMoney()` look similar, the only difference is that the first one searches by the id and the second one by card number. I suppose you can keep only one of them.

Reply:
deleted addBalance()